### PR TITLE
Make config loadable again

### LINKED ISSUE #21: Custom diktat config is currently ignored
We have an option to upload a custom config file, but it is ignored in code. It should be re-implemented.

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR closes #21 

### What's done:
- [x] Added `Upload config` button (again)
- [x] Added processing config files of backend
- [x] Added processing config files of frontend

### CODE REVIEW

File: src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt [hunk 1]
```diff
@@ -18,17 +18,37 @@ class DemoController {
...
+
+    /**
+     * @param configLines [String] of yaml config
+     * @param generatedName autogenerated name
+     * @return temp [File] that contains yaml config
+     */
+    fun prepareConfigFile(
+        configLines: String,
+        generatedName: String = generateFileName(),
+    ) = getDemoConfig(generatedName).apply {
```
Comment (petertrr) [inline src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt:32]: Should be `use{}` to guarantee closing?

File: src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt [hunk 2]
```diff
@@ -151,6 +153,31 @@ class EditorForm : RComponent<Props, CodeFormState>() {
                 }
             }
             br {}
+            div("row") {
+                setProp("align", "center")
+                div("upload-btn-wrapper") {
+                    button(classes = "btn") {
+                        +"Upload config"
+                        input(type = InputType.file, name = "diktat-analysis.yml") {
+                            attrs.accept = ".yml"
```
Comment (petertrr) [inline src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt:162]: What about `.yaml`?

File: src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt (hunk 1 above)
Comment (sanyavertolet) [reply src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt:32]: Yes, of course
Will fix it

File: src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt (hunk 2 above)
Comment (sanyavertolet) [reply src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt:162]: Will add it

File: src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt (hunk 1 above)
Comment (petertrr) [reply src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt]: Actually `writeText` creates an output stream and closes it all by itstlf, no need to chang anything

**Review by petertrr — CHANGES_REQUESTED** [review]
Please also add indication that the file has been uploaded and is currently active. No need to show more info about it, though; let's further improve it when we'll add interactive editing of config

File: src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt (hunk 2 above)
Comment (sanyavertolet) [reply src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt]: Done

File: src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt [hunk 3]
```diff
@@ -161,21 +157,37 @@ class EditorForm : RComponent<Props, CodeFormState>() {
...
+                                }
+                            }
                         }
                     }
+                    div {
+                         +"Loading..."
+                        attrs.hidden = !state.isLoading
+                    }
+                    div {
+                        +(state.codeForm.diktatConfig?.let{ "Config loaded" } ?: "")
```
Comment (petertrr) [inline src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt]: Maybe add `hidden` for this to?? Now you'll see an empty div. Or you could just unify those two, with conditional text

**Review by petertrr — APPROVED** [review]

File: src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt (hunk 3 above)
Comment (sanyavertolet) [reply src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt]: Seems like adding hidden is a good idea in order to make our `index.html` look a bit better

Comment (sanyavertolet) [reply src/jsMain/kotlin/org/cqfn/diktat/demo/frontend/components/EditorForm.kt]: Ok, will make it hideable

File: src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt [hunk 4]
```diff
@@ -18,17 +18,37 @@ class DemoController {
...
      * @param codeFormHtml
      * @return a page name
      */
+    @Suppress("TOO_LONG_FUNCTION")
     fun checkAndFixCode(codeFormHtml: CodeForm): CodeForm {
         val codeForm = codeFormHtml
-        val codeFix = CodeFix(codeForm.initialCode!!, codeFormHtml.ruleSet[0])
-        val file = getDemoFile()
+        val generatedName = generateFileName()
+        println(codeFormHtml.diktatConfig)
```
Comment (nulls) [inline src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt:44]: is it okay to have println here? (maybe left debug logs)

Comment (sanyavertolet) [reply src/jvmMain/kotlin/org/cqfn/diktat/demo/controller/DemoController.kt:44]: Sure it is. Will fix it

**Pull request merged**